packages/tracing-processing/src/services/enricherService.ts

```diff
@@ -70,36 +69,33 @@ export function dbServiceBuilder(db: DB) {
               );
 
               if (!consumer) {
-                return {
-                  purposeId: record.purpose_id,
-                  status: record.status,
-                  errorCode: PurposeErrorCodes.CONSUMER_NOT_FOUND,
-                  message: PurposeErrorCodes.CONSUMER_NOT_FOUND,
-                  rowNumber: record.rowNumber,
-                };
+                throw getEnrichedPurposeError(
+                  `Consumer ${tracing.tenantId} not found for tracingId: ${tracing.tracingId}, purpose_id: ${record.purpose_id}`,
+                );
               }
 
               if (!eService) {
```

Comment:
let's move this [here](https://github.com/pagopa/interop-tracing-core/pull/9/files/d0008b8e90281c526a5a927157f1023ccf06bfe7..51fb8afaa47a080f805b3f467fdca62f96f842cb#diff-7d6e39674e943de6a8bcf94e7d87f1436c8414239e4df9665bc7e6316abb4bf3R61), no need to do further queries if an error has been found